diff --git a/lib/matplotlib/tests/test_axes.py b/lib/matplotlib/tests/test_axes.py
--- a/lib/matplotlib/tests/test_axes.py
+++ b/lib/matplotlib/tests/test_axes.py
@@ -486,5 +486,16 @@ def test_inverted_cla():
     # clean up
     plt.close(fig)
 
 
+def test_clear_fully_removes_artist():
+    fig, ax = plt.subplots()
+    line, = ax.plot([0, 1])
+    ax.clear()
+
+    assert line.axes is None
+    assert line.figure is None
+    with pytest.raises(ValueError):
+        line.remove()
+
+
 def test_subclass_clear_cla():
