diff --git a/tests/forms_tests/field_tests/test_floatfield.py b/tests/forms_tests/field_tests/test_floatfield.py
--- a/tests/forms_tests/field_tests/test_floatfield.py
+++ b/tests/forms_tests/field_tests/test_floatfield.py
@@ -38,6 +38,11 @@ class FloatFieldTest(FormFieldAssertionsMixin, SimpleTestCase):
         with self.assertRaisesMessage(ValidationError, "'Enter a number.'"):
             f.clean('-Inf')
 
+    def test_non_finite_value_custom_error_message(self):
+        f = FloatField(error_messages={'invalid': '"%(value)s" is not a finite number.'})
+        with self.assertRaisesMessage(ValidationError, '"inf" is not a finite number.'):
+            f.clean(float('inf'))
+
     def test_floatfield_2(self):
         f = FloatField(required=False)
         self.assertIsNone(f.clean(''))
