diff --git a/tests/user_commands/test_command_parser.py b/tests/user_commands/test_command_parser.py
new file mode 100644
--- /dev/null
+++ b/tests/user_commands/test_command_parser.py
@@ -0,0 +1,19 @@
+from django.core.management.base import CommandError, CommandParser
+from django.test import SimpleTestCase
+
+
+class CommandParserTests(SimpleTestCase):
+    def test_subparser_preserves_programmatic_mode(self):
+        class CommandLineParser(CommandParser):
+            def __init__(self, *, called_from_command_line=True, **kwargs):
+                super().__init__(
+                    called_from_command_line=called_from_command_line, **kwargs
+                )
+
+        parser = CommandParser()
+        subparsers = parser.add_subparsers(parser_class=CommandLineParser)
+        create = subparsers.add_parser("create")
+        create.add_argument("name")
+        msg = "Error: the following arguments are required: name"
+        with self.assertRaisesMessage(CommandError, msg):
+            parser.parse_args(["create"])
