diff --git a/tests/messages_tests/tests.py b/tests/messages_tests/tests.py
--- a/tests/messages_tests/tests.py
+++ b/tests/messages_tests/tests.py
@@ -1,6 +1,7 @@
 from unittest import mock
 
 from django.contrib.messages import constants
+from django.contrib.messages.storage import base
 from django.contrib.messages.storage.base import Message
 from django.test import SimpleTestCase
 
@@ -15,3 +16,6 @@ class MessageTests(SimpleTestCase):
         self.assertNotEqual(msg_1, msg_2)
         self.assertNotEqual(msg_1, msg_3)
         self.assertNotEqual(msg_2, msg_3)
+
+    def test_storage_base_does_not_expose_signal(self):
+        self.assertFalse(hasattr(base, 'Signal'))
