diff --git a/lib/matplotlib/tests/test_ticker.py b/lib/matplotlib/tests/test_ticker.py
index ed61f9b11..ef2f3eed7 100644
--- a/lib/matplotlib/tests/test_ticker.py
+++ b/lib/matplotlib/tests/test_ticker.py
@@ -202,6 +202,14 @@ class TestLogLocator(object):
         loc = mticker.LogLocator(base=2)
         test_value = np.array([0.5, 1., 2., 4., 8., 16., 32., 64., 128., 256.])
         assert_almost_equal(loc.tick_values(1, 100), test_value)
+
+    def test_view_limits_reversed_with_nonpositive_endpoint(self):
+        fig, ax = plt.subplots()
+        ax.plot([10, 100], [1, 2])
+        ax.set_xscale('log')
+
+        locator = ax.xaxis.get_major_locator()
+        assert locator.view_limits(1, 0) == (1, 10)
 
     def test_switch_to_autolocator(self):
         loc = mticker.LogLocator(subs="all")
