diff --git a/tests/migrations/test_writer.py b/tests/migrations/test_writer.py
--- a/tests/migrations/test_writer.py
+++ b/tests/migrations/test_writer.py
@@ -433,6 +433,10 @@ class WriterTests(SimpleTestCase):
         Make sure compiled regex can be serialized.
         """
         regex = re.compile(r'^\w+$')
         self.assertSerializedEqual(regex)
 
+    def test_serialize_compiled_regex_with_combined_flags(self):
+        regex = re.compile(r'^foo$', re.IGNORECASE | re.MULTILINE)
+        self.assertSerializedEqual(regex)
+
     def test_serialize_class_based_validators(self):
