diff --git a/lib/matplotlib/tests/test_ticker.py b/lib/matplotlib/tests/test_ticker.py
--- a/lib/matplotlib/tests/test_ticker.py
+++ b/lib/matplotlib/tests/test_ticker.py
@@ -203,6 +203,16 @@ class TestLogLocator(object):
         test_value = np.array([0.5, 1., 2., 4., 8., 16., 32., 64., 128., 256.])
         assert_almost_equal(loc.tick_values(1, 100), test_value)
 
+    def test_nonsingular_reversed_nonpositive_bound(self):
+        fig, ax = plt.subplots()
+        ax.set_xscale('log')
+        ax.plot([0.1, 1, 10], [0, 1, 2])
+        locator = ax.xaxis.get_major_locator()
+
+        expected = (0.1, 10)
+        assert_array_equal(locator.nonsingular(-1, 10), expected)
+        assert_array_equal(locator.nonsingular(10, -1), expected)
+
     def test_switch_to_autolocator(self):
         loc = mticker.LogLocator(subs="all")
         assert_array_equal(loc.tick_values(0.45, 0.55),
