diff --git a/sympy/sets/tests/test_fancysets.py b/sympy/sets/tests/test_fancysets.py
--- a/sympy/sets/tests/test_fancysets.py
+++ b/sympy/sets/tests/test_fancysets.py
@@ -665,7 +665,14 @@ def test_imageset_intersect_real():
     # should be canonical
     assert s.intersect(S.Reals) == imageset(
         Lambda(n, 2*n*pi - pi/4), S.Integers) == ImageSet(
         Lambda(n, 2*pi*n + pi*Rational(7, 4)), S.Integers)
 
 
+def test_imageset_intersect_real_with_mixed_factors():
+    from sympy.abc import n
+    s = imageset(
+        Lambda(n, n + I*(n - 1)*(n + 1)*(n**2 + 1)), S.Integers)
+    assert s.intersect(S.Reals) == FiniteSet(-1, 1)
+
+
 def test_imageset_intersect_interval():
