diff --git a/sympy/printing/tests/test_str.py b/sympy/printing/tests/test_str.py
index 690b1a8bbf..c73b9f621e 100644
--- a/sympy/printing/tests/test_str.py
+++ b/sympy/printing/tests/test_str.py
@@ -269,5 +269,12 @@ def test_Mul():
     assert str(cc1*Rational(2)*cc2) == '2*CustomClass1()*CustomClass2()'
 
 
+def test_Mul_reciprocal_of_positive_power():
+    reciprocal_of_square = Pow(Pow(y, 2, evaluate=False), -1,
+                               evaluate=False)
+    expr = Mul(x, reciprocal_of_square, evaluate=False)
+    assert str(expr) == str(x/y**2) == 'x/y**2'
+
+
 def test_NaN():
     assert str(nan) == "nan"
