diff --git a/tests/test_domain_py.py b/tests/test_domain_py.py
--- a/tests/test_domain_py.py
+++ b/tests/test_domain_py.py
@@ -983,6 +983,16 @@ def test_info_field_list(app):
                 refdomain="py", reftype="class", reftarget="str",
                 **{"py:module": "example", "py:class": "Class"})
 
 
+def test_info_field_list_type_containing_quoted_bracket(app):
+    text = (".. py:function:: example(bracket)\n"
+            "\n"
+            '   :param Literal["["] bracket: opening bracket\n')
+    doctree = restructuredtext.parse(app, text)
+
+    field = doctree.traverse(nodes.field)[0]
+    assert field[1].astext() == 'bracket (Literal["["]) -- opening bracket'
+
+
 def test_info_field_list_var(app):
     text = (".. py:class:: Class\n"
             "\n"
