diff --git a/sympy/core/tests/test_arit.py b/sympy/core/tests/test_arit.py
--- a/sympy/core/tests/test_arit.py
+++ b/sympy/core/tests/test_arit.py
@@ -1663,6 +1663,11 @@ def test_Mod():
     assert Mod(Mod(x + 2, 4)*4, 4) == 0
 
 
+def test_Mod_even_dividend_quotient():
+    e = Symbol('e', even=True)
+    assert Mod(e/2, 2).subs(e, 6) == Mod(3, 2)
+
+
 def test_Mod_is_integer():
     p = Symbol('p', integer=True)
     q1 = Symbol('q1', integer=True)
