diff --git a/tests/utils_tests/test_http.py b/tests/utils_tests/test_http.py
--- a/tests/utils_tests/test_http.py
+++ b/tests/utils_tests/test_http.py
@@ -1,5 +1,6 @@
 import unittest
 from datetime import datetime
+from unittest import mock
 
 from django.test import SimpleTestCase, ignore_warnings
 from django.utils.datastructures import MultiValueDict
@@ -319,7 +320,17 @@ class HttpDateProcessingTests(unittest.TestCase):
     def test_parsing_rfc850(self):
         parsed = parse_http_date('Sunday, 06-Nov-94 08:49:37 GMT')
         self.assertEqual(datetime.utcfromtimestamp(parsed), datetime(1994, 11, 6, 8, 49, 37))
 
+    def test_parsing_rfc850_year_exactly_50_years_in_future(self):
+        class MockDateTime(datetime):
+            @classmethod
+            def utcnow(cls):
+                return cls(1969, 1, 1)
+
+        with mock.patch('django.utils.http.datetime.datetime', MockDateTime):
+            parsed = parse_http_date('Tuesday, 01-Jan-19 00:00:00 GMT')
+        self.assertEqual(datetime.utcfromtimestamp(parsed), datetime(2019, 1, 1))
+
     def test_parsing_asctime(self):
         parsed = parse_http_date('Sun Nov  6 08:49:37 1994')
         self.assertEqual(datetime.utcfromtimestamp(parsed), datetime(1994, 11, 6, 8, 49, 37))
