diff --git a/tests/i18n/patterns/tests.py b/tests/i18n/patterns/tests.py
index ebd2430428..3c144598b5 100644
--- a/tests/i18n/patterns/tests.py
+++ b/tests/i18n/patterns/tests.py
@@ -76,6 +76,17 @@ class URLPrefixTests(URLTestCaseBase):
         with translation.override(None):
             self.assertEqual(reverse('prefixed'), '/%s/prefixed/' % settings.LANGUAGE_CODE)
 
+    @override_settings(
+        LANGUAGE_CODE='de',
+        LANGUAGES=[
+            ('de', 'German'),
+            ('en-latn-us-posix', 'Latin English (POSIX)'),
+        ],
+    )
+    def test_language_prefix_with_variant(self):
+        response = self.client.get('/en-latn-us-posix/prefixed/')
+        self.assertEqual(response.status_code, 200)
+
     @override_settings(ROOT_URLCONF='i18n.patterns.urls.wrong')
     def test_invalid_prefix_use(self):
         msg = 'Using i18n_patterns in an included URLconf is not allowed.'
