diff --git a/tests/model_enums/tests.py b/tests/model_enums/tests.py
--- a/tests/model_enums/tests.py
+++ b/tests/model_enums/tests.py
@@ -233,6 +233,9 @@ class CustomChoicesTests(SimpleTestCase):
         for choice_enum in enums:
             with self.subTest(choice_enum.__name__):
                 self.assertNotIn(None, choice_enum.labels)
 
+    def test_str(self):
+        self.assertEqual(str(MoonLandings.APOLLO_11), '1969-07-20')
+
     def test_bool_unsupported(self):
         msg = "type 'bool' is not an acceptable base type"
         with self.assertRaisesMessage(TypeError, msg):
