diff --git a/tests/test_util_inspect.py b/tests/test_util_inspect.py
--- a/tests/test_util_inspect.py
+++ b/tests/test_util_inspect.py
@@ -10,6 +10,7 @@
 
 import ast
 import datetime
+import enum
 import functools
 import sys
 import types
@@ -62,6 +63,21 @@ def test_signature():
     assert sig == '(a, b, c=1, d=2, *e, **f)'
 
 
+def test_stringify_signature_enum_with_overridden_name():
+    class CustomNameEnum(enum.Enum):
+        VALUE = 1
+
+        @property
+        def name(self):
+            return 'overridden'
+
+    def function(value=CustomNameEnum.VALUE):
+        pass
+
+    sig = inspect.signature(function)
+    assert stringify_signature(sig) == '(value=CustomNameEnum.VALUE)'
+
+
 def test_signature_partial():
     def fun(a, b, c=1, d=2):
         pass
