diff --git a/tests/roots/test-prolog/restructuredtext.rst b/tests/roots/test-prolog/restructuredtext.rst
--- a/tests/roots/test-prolog/restructuredtext.rst
+++ b/tests/roots/test-prolog/restructuredtext.rst
@@ -1,3 +1,5 @@
+:author: The ``Sphinx`` team
+
 sample document
 ===============
 
diff --git a/tests/test_markup.py b/tests/test_markup.py
--- a/tests/test_markup.py
+++ b/tests/test_markup.py
@@ -539,6 +539,12 @@ def test_rst_prolog(app, status, warning):
     # rst_prolog & rst_epilog on exlucding reST parser
     assert not md.rawsource.startswith('*Hello world*.')
     assert not md.rawsource.endswith('*Good-bye world*.\n')
+
+
+@pytest.mark.sphinx('dummy', testroot='prolog')
+def test_rst_prolog_after_docinfo_with_inline_markup(app):
+    app.builder.build_all()
+    assert app.env.metadata['restructuredtext']['author'] == 'The Sphinx team'
 
 
 @pytest.mark.sphinx('dummy', testroot='keep_warnings')
