diff --git a/sympy/sets/tests/test_fancysets.py b/sympy/sets/tests/test_fancysets.py
--- a/sympy/sets/tests/test_fancysets.py
+++ b/sympy/sets/tests/test_fancysets.py
@@ -1136,6 +1136,12 @@ def test_imageset_intersection():
         log(Abs(sqrt(-I))))), S.Integers)
     assert s.intersect(S.Reals) == ImageSet(
         Lambda(n, 2*pi*n + pi*Rational(7, 4)), S.Integers)
+
+
+def test_imageset_reals_nonlinear_imaginary_part():
+    n = Symbol('n')
+    s = imageset(Lambda(n, n + I*(n**2 - 4)), S.Integers)
+    assert s.intersect(S.Reals) == FiniteSet(-2, 2)
 
 
 def test_issue_17858():
