diff --git a/tests/messages_tests/tests.py b/tests/messages_tests/tests.py
--- a/tests/messages_tests/tests.py
+++ b/tests/messages_tests/tests.py
@@ -1,8 +1,8 @@
 from unittest import mock
 
 from django.contrib.messages import constants
-from django.contrib.messages.storage.base import Message
-from django.test import SimpleTestCase
+from django.contrib.messages.storage.base import LEVEL_TAGS, Message
+from django.test import SimpleTestCase, override_settings
 
 
 class MessageTests(SimpleTestCase):
@@ -15,3 +15,7 @@ class MessageTests(SimpleTestCase):
         self.assertNotEqual(msg_1, msg_2)
         self.assertNotEqual(msg_1, msg_3)
         self.assertNotEqual(msg_2, msg_3)
+
+    @override_settings(MESSAGE_TAGS={constants.INFO: 'custom-info'})
+    def test_imported_level_tags_track_override(self):
+        self.assertEqual(LEVEL_TAGS[constants.INFO], 'custom-info')
