diff --git a/tests/migrations/test_optimizer.py b/tests/migrations/test_optimizer.py
--- a/tests/migrations/test_optimizer.py
+++ b/tests/migrations/test_optimizer.py
@@ -203,7 +203,18 @@ class OptimizerTests(SimpleTestCase):
     def test_alter_alter_unique_model(self):
         self._test_alter_alter_model(
             migrations.AlterUniqueTogether("Foo", [["a", "b"]]),
             migrations.AlterUniqueTogether("Foo", [["a", "c"]]),
         )
 
+    def test_alter_unique_together_custom_subclass(self):
+        class CustomAlterUniqueTogether(migrations.AlterUniqueTogether):
+            pass
+
+        custom_operation = CustomAlterUniqueTogether("Foo", [["a", "b"]])
+        operation = migrations.AlterUniqueTogether("Foo", [["a", "c"]])
+        result = MigrationOptimizer().optimize(
+            [custom_operation, operation], "migrations",
+        )
+        self.assertEqual(result, [custom_operation, operation])
+
     def test_alter_alter_index_model(self):
         self._test_alter_alter_model(
             migrations.AlterIndexTogether("Foo", [["a", "b"]]),
