diff --git a/tests/i18n/tests.py b/tests/i18n/tests.py
index 0093181f64..3bf906f62d 100644
--- a/tests/i18n/tests.py
+++ b/tests/i18n/tests.py
@@ -1651,6 +1651,9 @@ class FormattingTests(SimpleTestCase):
 
     def test_format_arbitrary_settings(self):
         self.assertEqual(get_format("DEBUG"), "DEBUG")
+
+    def test_get_format_bytes_uses_str_coercion(self):
+        self.assertEqual(get_format(b"DEBUG", use_l10n=False), str(b"DEBUG"))
 
     def test_get_custom_format(self):
         reset_format_cache()
