diff --git a/tests/backends/base/test_operations.py b/tests/backends/base/test_operations.py
--- a/tests/backends/base/test_operations.py
+++ b/tests/backends/base/test_operations.py
@@ -65,6 +65,12 @@ class SimpleDatabaseOperationTests(SimpleTestCase):
         value = timezone.now().date()
         self.assertEqual(self.ops.adapt_unknown_value(value), self.ops.adapt_datefield_value(value))
 
+    def test_iso_year_lookup_bounds_for_date_field(self):
+        self.assertEqual(
+            self.ops.iso_year_lookup_bounds_for_date_field(2015),
+            ['2014-12-29', '2016-01-03'],
+        )
+
     def test_adapt_unknown_value_time(self):
         value = timezone.now().time()
         self.assertEqual(self.ops.adapt_unknown_value(value), self.ops.adapt_timefield_value(value))
