diff --git a/tests/config/test_argparse_config.py b/tests/config/test_argparse_config.py
--- a/tests/config/test_argparse_config.py
+++ b/tests/config/test_argparse_config.py
@@ -31,6 +31,17 @@ class TestArgparseOptionsProviderMixin:
         with pytest.raises(SystemExit) as ex:
             Run([LOGGING_TEST])
         assert ex.value.code == 2
 
+    @staticmethod
+    def test_ignore_paths_regex_with_comma_in_quantifier() -> None:
+        """Check that a comma in a path regex quantifier is not a separator."""
+        args = [
+            LOGGING_TEST,
+            r"--ignore-paths=.*log{1,3}ing_format_interpolation_style.py$",
+        ]
+        with pytest.raises(SystemExit) as ex:
+            Run(args)
+        assert ex.value.code == 0
+
     @staticmethod
     def test_logger_commandline() -> None:
