diff --git a/lib/matplotlib/tests/test_ticker.py b/lib/matplotlib/tests/test_ticker.py
--- a/lib/matplotlib/tests/test_ticker.py
+++ b/lib/matplotlib/tests/test_ticker.py
@@ -203,6 +203,12 @@ class TestLogLocator(object):
         test_value = np.array([0.5, 1., 2., 4., 8., 16., 32., 64., 128., 256.])
         assert_almost_equal(loc.tick_values(1, 100), test_value)
 
+    def test_nonsingular_no_positive_without_axis(self):
+        loc = mticker.LogLocator()
+        with pytest.warns(UserWarning, match="Data has no positive values"):
+            limits = loc.nonsingular(-2, -1)
+        assert limits == (1, 10)
+
     def test_switch_to_autolocator(self):
         loc = mticker.LogLocator(subs="all")
         assert_array_equal(loc.tick_values(0.45, 0.55),
