diff --git a/tests/fixtures_regress/tests.py b/tests/fixtures_regress/tests.py
--- a/tests/fixtures_regress/tests.py
+++ b/tests/fixtures_regress/tests.py
@@ -555,6 +555,20 @@ class TestFixtures(TestCase):
         ):
             management.call_command("loaddata", "absolute.json", verbosity=0)
 
+    def test_fixture_dirs_with_pathlike_duplicates(self):
+        fixture_dir = os.path.join(_cur_dir, "fixtures_1")
+
+        class FixturePath(os.PathLike):
+            def __fspath__(self):
+                return fixture_dir
+
+        with override_settings(FIXTURE_DIRS=[fixture_dir, FixturePath()]):
+            with self.assertRaisesMessage(
+                ImproperlyConfigured,
+                "settings.FIXTURE_DIRS contains duplicates.",
+            ):
+                management.call_command("loaddata", "absolute.json", verbosity=0)
+
     @override_settings(FIXTURE_DIRS=[os.path.join(_cur_dir, "fixtures")])
     def test_fixture_dirs_with_default_fixture_path(self):
         """
