diff --git a/tests/forms_tests/tests/test_forms.py b/tests/forms_tests/tests/test_forms.py
--- a/tests/forms_tests/tests/test_forms.py
+++ b/tests/forms_tests/tests/test_forms.py
@@ -2076,6 +2076,18 @@ class FormsTestCase(SimpleTestCase):
 
         f = SampleForm()
         self.assertEqual(f['name'], (f, 'name'))
 
+    def test_filefield_clean_custom_boundfield(self):
+        class CustomFileField(FileField):
+            def get_bound_field(self, form, name):
+                return (form, name)
+
+        class FileForm(Form):
+            file1 = CustomFileField(initial='resume.txt')
+
+        form = FileForm({})
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.cleaned_data['file1'], 'resume.txt')
+
     def test_initial_datetime_values(self):
         now = datetime.datetime.now()
         # Nix microseconds (since they should be ignored). #22502
