diff --git a/sympy/core/tests/test_evalf.py b/sympy/core/tests/test_evalf.py
--- a/sympy/core/tests/test_evalf.py
+++ b/sympy/core/tests/test_evalf.py
@@ -328,6 +328,14 @@ def test_implemented_function_evalf():
     del f._imp_     # XXX: due to caching _imp_ would influence all other tests
 
 
+def test_implemented_function_evalf_symbolic_result():
+    from sympy.utilities.lambdify import implemented_function
+    f = Function('symbolic_result_evalf')
+    f = implemented_function(f, lambda x: x*I)
+    assert f(2).evalf() == (2*I).evalf()
+    del f._imp_
+
+
 def test_evaluate_false():
     for no in [0, False]:
         assert Add(3, 2, evaluate=no).is_Add
