diff --git a/tests/model_fields/tests.py b/tests/model_fields/tests.py
--- a/tests/model_fields/tests.py
+++ b/tests/model_fields/tests.py
@@ -250,3 +250,17 @@ class GetChoicesOrderingTests(TestCase):
             self.field.remote_field.get_choices(include_blank=False, ordering=('-a',)),
             [self.bar2, self.bar1]
         )
+
+    def test_reverse_get_choices_with_meta_ordering_none(self):
+        original_ordering = Bar._meta.ordering
+        Bar._meta.ordering = None
+        try:
+            self.assertCountEqual(
+                self.field.remote_field.get_choices(include_blank=False),
+                [
+                    (self.bar1.pk, str(self.bar1)),
+                    (self.bar2.pk, str(self.bar2)),
+                ],
+            )
+        finally:
+            Bar._meta.ordering = original_ordering
