diff --git a/sympy/codegen/tests/test_ast.py b/sympy/codegen/tests/test_ast.py
--- a/sympy/codegen/tests/test_ast.py
+++ b/sympy/codegen/tests/test_ast.py
@@ -279,6 +279,11 @@ def test_String():
     assert str(s) == 'foo'
     assert repr(s) == "String('foo')"
 
+def test_String_args_do_not_make_text_symbolic():
+    s = String('foo')
+    assert s.func(*s.args) == s
+    assert not s.has(Symbol('foo'))
+
 def test_Comment():
     c = Comment('foobar')
     assert c.text == 'foobar'
