diff --git a/tests/migrations/test_writer.py b/tests/migrations/test_writer.py
index 622c79c3db..be74cc7b75 100644
--- a/tests/migrations/test_writer.py
+++ b/tests/migrations/test_writer.py
@@ -251,6 +251,25 @@ class WriterTests(SimpleTestCase):
         lazy_pattern = SimpleLazyObject(lambda: pattern)
         self.assertEqual(self.serialize_round_trip(lazy_pattern), pattern)
 
+    def test_serialize_enum_with_dynamic_name(self):
+        class RenamingEnum(enum.Enum):
+            VALUE = 'value'
+
+            @property
+            def name(self):
+                self.__class__.__name__ = 'RenamedEnum'
+                return self._name_
+
+        globals()['RenamingEnum'] = RenamingEnum
+        try:
+            self.assertIs(
+                self.serialize_round_trip(RenamingEnum.VALUE),
+                RenamingEnum.VALUE,
+            )
+        finally:
+            RenamingEnum.__name__ = 'RenamingEnum'
+            del globals()['RenamingEnum']
+
     def test_serialize_enums(self):
         class TextEnum(enum.Enum):
             A = 'a-value'
