diff --git a/tests/i18n/tests.py b/tests/i18n/tests.py
--- a/tests/i18n/tests.py
+++ b/tests/i18n/tests.py
@@ -1651,6 +1651,9 @@ class FormattingTests(SimpleTestCase):
 
     def test_format_arbitrary_settings(self):
         self.assertEqual(get_format("DEBUG"), "DEBUG")
+
+    def test_get_format_bytes(self):
+        self.assertEqual(get_format(b"Y-m-d", use_l10n=False), "Y-m-d")
 
     def test_get_custom_format(self):
         reset_format_cache()
