diff --git a/tests/test_util_inspect.py b/tests/test_util_inspect.py
index 9d1f52d4d..040a4c19d 100644
--- a/tests/test_util_inspect.py
+++ b/tests/test_util_inspect.py
@@ -10,6 +10,7 @@
 
 import ast
 import datetime
+import enum
 import functools
 import sys
 import types
@@ -64,6 +65,25 @@ def test_signature():
 
     sig = inspect.stringify_signature(inspect.signature(func))
     assert sig == '(a, b, c=1, d=2, *e, **f)'
 
 
+def test_signature_enum_default_with_formattable_name():
+    class FormattedName(str):
+        def __format__(self, format_spec):
+            return 'ValueA'
+
+    class Choice(enum.Enum):
+        ValueA = 10
+
+        @property
+        def name(self):
+            return FormattedName('internal-name')
+
+    def func(value=Choice.ValueA):
+        pass
+
+    signature = stringify_signature(inspect.signature(func))
+    assert signature == '(value=Choice.ValueA)'
+
+
 def test_signature_partial():
     def fun(a, b, c=1, d=2):
         pass
