diff --git a/tests/test_build_linkcheck.py b/tests/test_build_linkcheck.py
--- a/tests/test_build_linkcheck.py
+++ b/tests/test_build_linkcheck.py
@@ -157,3 +157,10 @@ def test_linkcheck_request_headers(app, status, warning):
                 assert headers["X-Secret"] == "open sesami"
             else:
                 assert headers["Accept"] == "text/html,application/xhtml+xml;q=0.9,*/*;q=0.8"
+
+
+def test_linkcheck_star_import_does_not_expose_os():
+    namespace = {}
+    exec('from sphinx.builders.linkcheck import *', namespace)
+
+    assert 'os' not in namespace
