diff --git a/tests/forms_tests/tests/test_forms.py b/tests/forms_tests/tests/test_forms.py
--- a/tests/forms_tests/tests/test_forms.py
+++ b/tests/forms_tests/tests/test_forms.py
@@ -2496,11 +2496,14 @@ class FormsTestCase(SimpleTestCase):
         )
 
     def test_filefield_initial_callable(self):
+        initial_values = iter(['resume.txt', 'alternate.txt'])
+
         class FileForm(forms.Form):
-            file1 = forms.FileField(initial=lambda: 'resume.txt')
+            file1 = forms.FileField(initial=lambda: next(initial_values))
 
         f = FileForm({})
+        initial = f['file1'].initial
         self.assertEqual(f.errors, {})
-        self.assertEqual(f.cleaned_data['file1'], 'resume.txt')
+        self.assertEqual(f.cleaned_data['file1'], initial)
 
     def test_filefield_with_fileinput_required(self):
