diff --git a/tests/validators/tests.py b/tests/validators/tests.py
--- a/tests/validators/tests.py
+++ b/tests/validators/tests.py
@@ -328,6 +328,18 @@ class TestValidators(SimpleTestCase):
                         validator(value)
                 else:
                     self.assertEqual(expected, validator(value))
+
+    def test_decimal_validator_non_finite_value_error_code(self):
+        value = Decimal('NaN')
+        validator = DecimalValidator(max_digits=10, decimal_places=2)
+        validator.messages = {
+            'invalid': '"%(value)s" is not a number.',
+        }
+        with self.assertRaises(ValidationError) as cm:
+            validator(value)
+        self.assertEqual(cm.exception.messages, ['"NaN" is not a number.'])
+        self.assertIs(cm.exception.params['value'], value)
+        self.assertIsNone(cm.exception.code)
 
     def test_single_message(self):
         v = ValidationError('Not Valid')
