diff --git a/tests/i18n/tests.py b/tests/i18n/tests.py
--- a/tests/i18n/tests.py
+++ b/tests/i18n/tests.py
@@ -518,6 +518,9 @@ class TranslationThreadSafetyTests(SimpleTestCase):
         self.assertLess(translation_count, len(trans_real._translations))
 
 
 class FormattingTests(SimpleTestCase):
+    def test_get_format_has_no_docstring(self):
+        self.assertIsNone(get_format.__doc__)
+
     def setUp(self):
         super().setUp()
         self.n = decimal.Decimal("66666.666")
