diff --git a/tests/utils_tests/test_autoreload.py b/tests/utils_tests/test_autoreload.py
--- a/tests/utils_tests/test_autoreload.py
+++ b/tests/utils_tests/test_autoreload.py
@@ -10,6 +10,7 @@ import types
 import weakref
 import zipfile
 from importlib import import_module
+from importlib.machinery import ModuleSpec
 from pathlib import Path
 from subprocess import CompletedProcess
 from unittest import mock, skip, skipIf
@@ -165,6 +166,16 @@ class TestChildArguments(SimpleTestCase):
             [sys.executable, '-m', 'django', 'runserver']
         )
 
+    @mock.patch('__main__.__spec__', ModuleSpec('custom_cli.__main__', loader=None))
+    @mock.patch('__main__.__file__', None)
+    @mock.patch('sys.argv', [__file__, 'runserver'])
+    @mock.patch('sys.warnoptions', [])
+    def test_run_as_module_without_main_file(self):
+        self.assertEqual(
+            autoreload.get_child_arguments(),
+            [sys.executable, '-m', 'custom_cli', 'runserver']
+        )
+
     @mock.patch('sys.argv', [__file__, 'runserver'])
     @mock.patch('sys.warnoptions', ['error'])
     def test_warnoptions(self):
