diff --git a/tests/backends/base/test_operations.py b/tests/backends/base/test_operations.py
--- a/tests/backends/base/test_operations.py
+++ b/tests/backends/base/test_operations.py
@@ -1,3 +1,4 @@
+import datetime
 import decimal
 
 from django.core.management.color import no_style
@@ -17,6 +18,12 @@ class SimpleDatabaseOperationTests(SimpleTestCase):
     may_require_msg = 'subclasses of BaseDatabaseOperations may require a %s() method'
 
     def setUp(self):
         self.ops = BaseDatabaseOperations(connection=connection)
 
+    def test_iso_year_lookup_bounds_for_date_field(self):
+        self.assertEqual(
+            self.ops.iso_year_lookup_bounds_for_date_field(2020),
+            [datetime.date(2019, 12, 30), datetime.date(2021, 1, 3)],
+        )
+
     def test_deferrable_sql(self):
         self.assertEqual(self.ops.deferrable_sql(), '')
