diff --git a/sympy/polys/matrices/tests/test_normalforms.py b/sympy/polys/matrices/tests/test_normalforms.py
--- a/sympy/polys/matrices/tests/test_normalforms.py
+++ b/sympy/polys/matrices/tests/test_normalforms.py
@@ -73,3 +73,8 @@ def test_hermite_normal():
     raises(DMDomainError, lambda: hermite_normal_form(m))
     raises(DMDomainError, lambda: _hermite_normal_form(m))
     raises(DMDomainError, lambda: _hermite_normal_form_modulo_D(m, ZZ(1)))
+
+
+def test_hermite_normal_zero_columns():
+    m = DM([[], [], []], ZZ)
+    assert hermite_normal_form(m) == m
