diff --git a/xarray/tests/test_duck_array_ops.py b/xarray/tests/test_duck_array_ops.py
--- a/xarray/tests/test_duck_array_ops.py
+++ b/xarray/tests/test_duck_array_ops.py
@@ -595,6 +595,14 @@ def test_min_count(dim_num, dtype, dask, func, aggdim):
     assert_dask_array(actual, dask)
 
 
+def test_sum_min_count_object_all_dimensions():
+    da = DataArray(
+        np.array([[1, np.nan], [np.nan, np.nan]], dtype=object), dims=("x", "y")
+    )
+
+    assert np.isnan(da.sum(skipna=True, min_count=2).item())
+
+
 @pytest.mark.parametrize("func", ["sum", "prod"])
 def test_min_count_dataset(func):
     da = construct_dataarray(2, dtype=float, contains_nan=True, dask=False)
