diff --git a/tests/i18n/tests.py b/tests/i18n/tests.py
--- a/tests/i18n/tests.py
+++ b/tests/i18n/tests.py
@@ -1864,6 +1864,14 @@ class CountrySpecificLanguageTests(SimpleTestCase):
         self.assertFalse(check_for_language('tr-TR.UTF-8'))
         self.assertFalse(check_for_language('tr-TR.UTF8'))
         self.assertFalse(check_for_language('de-DE.utf-8'))
 
+    @override_settings(LANGUAGES=[('sr-RS@latin', 'Serbian Latin')])
+    def test_get_language_from_request_path_gettext_modifier(self):
+        request = self.rf.get('/sr-RS@latin/')
+        self.assertEqual(
+            get_language_from_request(request, check_path=True),
+            'sr-RS@latin',
+        )
+
     def test_check_for_language_null(self):
         self.assertIs(trans_null.check_for_language('en'), True)
