diff --git a/tests/migrations/test_optimizer.py b/tests/migrations/test_optimizer.py
--- a/tests/migrations/test_optimizer.py
+++ b/tests/migrations/test_optimizer.py
@@ -7,5 +7,9 @@
 from .models import EmptyManager, UnicodeModel
 
 
+class CustomAlterUniqueTogether(migrations.AlterUniqueTogether):
+    pass
+
+
 class OptimizerTests(SimpleTestCase):
     """
@@ -203,7 +207,14 @@ class OptimizerTests(SimpleTestCase):
     def test_alter_alter_unique_model(self):
         self._test_alter_alter_model(
             migrations.AlterUniqueTogether("Foo", [["a", "b"]]),
             migrations.AlterUniqueTogether("Foo", [["a", "c"]]),
         )
 
+    def test_alter_alter_unique_model_subclass(self):
+        subclass_operation = CustomAlterUniqueTogether("Foo", [["a", "c"]])
+        self._test_alter_alter_model(
+            migrations.AlterUniqueTogether("Foo", [["a", "b"]]),
+            subclass_operation,
+        )
+
     def test_alter_alter_index_model(self):
