diff --git a/tests/filtered_relation/tests.py b/tests/filtered_relation/tests.py
--- a/tests/filtered_relation/tests.py
+++ b/tests/filtered_relation/tests.py
@@ -147,6 +147,17 @@ class FilteredRelationTests(TestCase):
             [self.author2]
         )
 
+    def test_exclude_with_outer_field_reference(self):
+        other_author = Author.objects.create(name=self.book1.title)
+        queryset = Author.objects.annotate(
+            same_title_book=FilteredRelation(
+                'book', condition=Q(book__title=F('name')),
+            ),
+        ).exclude(same_title_book__isnull=False).order_by('pk')
+        self.assertSequenceEqual(
+            queryset, [self.author1, self.author2, other_author],
+        )
+
     def test_with_m2m(self):
         qs = Author.objects.annotate(
             favorite_books_written_by_jane=FilteredRelation(
