diff --git a/tests/messages_tests/tests.py b/tests/messages_tests/tests.py
--- a/tests/messages_tests/tests.py
+++ b/tests/messages_tests/tests.py
@@ -15,3 +15,9 @@ class MessageTests(SimpleTestCase):
         self.assertNotEqual(msg_1, msg_2)
         self.assertNotEqual(msg_1, msg_3)
         self.assertNotEqual(msg_2, msg_3)
+
+    def test_setting_changed_signal_class(self):
+        from django.contrib.messages.storage.base import Signal
+        from django.core.signals import setting_changed
+
+        self.assertIs(type(setting_changed), Signal)
